### PR TITLE
Added Intros to REST API pages

### LINKED ISSUE #6978: Add intros to REST API pages
### What article on docs.github.com is affected?

- "[SCIM](https://docs.github.com/en/rest/reference/scim)"
- "[Search](https://docs.github.com/en/rest/reference/search)"
- "[Secret scanning](https://docs.github.com/en/rest/reference/secret-scanning)"
- "[Teams](https://docs.github.com/en/rest/reference/teams)"
- "[Users](https://docs.github.com/en/rest/reference/users)"

### What part(s) of the article would you like to see updated?

Each of these articles needs an intro to help orient users to the content on the page and what each API can be used for. Intros should be one sentence long and define what the feature is.

For "[Search](https://docs.github.com/en/rest/reference/search)" and "[Secret scanning](https://docs.github.com/en/rest/reference/secret-scanning)," you can cut and paste the first sentence in the article for the intro.

For "[SCIM](https://docs.github.com/en/rest/reference/scim)," "[Teams](https://docs.github.com/en/rest/reference/teams)," and "[Users](https://docs.github.com/en/rest/reference/users)" you will need to write new intros.

For more information about writing intros, see:

- [Information about writing intros](https://github.com/github/docs/blob/main/contributing/content-model.md#intro) in the GitHub docs content model
- [Information about the `intro` frontmatter property](https://github.com/github/docs/tree/main/content#intro) in the GitHub Docs content directory README

### Additional information

This issue is for the Campus Experts + Docs program and not currently available for contributors outside the program to work on.

### PR DESCRIPTION
**Pull request by codewithdev**
### Why:

Closes #6978 

### What's being changed:

Added Intros for the following REST API pages,
 -  "[SCIM](https://docs.github.com/en/rest/reference/scim)"
 -  "[Search](https://docs.github.com/en/rest/reference/search)"
 -  "[Secret scanning](https://docs.github.com/en/rest/reference/secret-scanning)"
 -  "[Teams](https://docs.github.com/en/rest/reference/teams)"
 -  "[Users](https://docs.github.com/en/rest/reference/users)"
 
### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by janiceilene** [comment]
🙌 Thanks so much for opening _another_ PR @codewithdev! I'll get this triaged for review ⚡

File: content/rest/reference/teams.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Teams
+intro: 'With Teams API you can create and manage teams in your {% data variables.product.product_name %} organization.'
```
Comment (runleonarun) [inline content/rest/reference/teams.md:3]:
```suggestion
intro: 'With the Teams API, you can create and manage teams in your {% data variables.product.product_name %} organization.'
```

File: content/rest/reference/search.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Search
+intro: 'The {% data variables.product.product_name %} search API lets you to search for the specific item efficiently.'
```
Comment (runleonarun) [inline content/rest/reference/search.md:3]:
```suggestion
intro: 'The {% data variables.product.product_name %} Search API lets you to search for the specific item efficiently.'
```

File: content/rest/reference/secret-scanning.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Secret scanning
+intro: 'To retrieve and update the secret alerts from a private repository, you can use secret scanning API.'
```
Comment (runleonarun) [inline content/rest/reference/secret-scanning.md:3]:
```suggestion
intro: 'To retrieve and update the secret alerts from a private repository, you can use Secret Scanning API.'
```

**Review by runleonarun — DISMISSED** [review]
Nice work @codewithdev! I just made a few tiny suggestions and then this is ready to merge! I know we are inconsistent with this, but we like to see the name of the APIs capitalized.

**Comment by codewithdev** [comment]
> Nice work @codewithdev! I just made a few tiny suggestions and then this is ready to merge! I know we are inconsistent with this, but we like to see the name of the APIs capitalized. 

Thanks, I will make sure to capitalise it for other REST API PR's as well.

**Review by janiceilene — APPROVED** [review]
Thanks for incorporating those suggestions ⚡

**Pull request merged**